pkg/build/builder/source_test.go

```diff
@@ -0,0 +1,26 @@
+package builder
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestCheckRemoteGit(t *testing.T) {
+	var err error
+	err = checkRemoteGit("https://github.com/openshift/private-repo.git", 10*time.Second)
+	switch v := err.(type) {
+	case gitAuthError:
+	default:
+		t.Errorf("expected gitAuthError, got %q", v)
+	}
+	err = checkRemoteGit("https://github.com/openshift/origin", 10*time.Second)
```

Comment:
let's drop this... not sure I want another unit test requiring connectivity to github :-/